User Management/src/test/java/com/usermanagement/Course/CourseManagement/CreateCourseTests.java

```diff
@@ -0,0 +1,197 @@
+	 * 3.- Course insert not successfull when inserting 2 or more categories that doesnt exists
+	 * 4.- Course insert not successfull when inserting a course with a repeated name
+	 * 5.- Course insert not successfull when inserting a course with no categories included
+	 * 6.- Course insert not successfull when sending an course with an empty name and content
+	 * 7.- Course insert not successfull when inserting a course with a name longer than 50 characters
+	 * 8.- Course insert not successfull when inserting a coruse with a description longer than 500 characters
+	 * 9.- Course insert not successfull when inserting a course with a name with special chars
+	 * 
+	 */
+	
+	@Test
+	public void insertCourseSuccessfully() {
+		String name = "A course title";
+		String description = "A course desription";
+		Integer[] categories = {1,2,3};
+		String content = "A course content";
+
+		CourseRequest course = new CourseRequest();
+		course.setName(name);
+		course.setDescription(description);
+		course.setCategories(categories);
+		course.setContent(content);
+
+		when(courseResourceMock.createCourse(course)).thenReturn(new ResponseEntity<>(HttpStatus.CREATED));
+		
+		assertEquals(HttpStatus.CREATED, courseResourceMock.createCourse(course).getStatusCode());
+	}
+	
+	@Test
+	public void insertCourseSuccessfullyWithEmptyDescription() {
+		String name = "A course title";
+		String description = "";
+		Integer[] categories = {1,2,3};
+		String content = "A course content";
+
+		CourseRequest course = new CourseRequest();
+		course.setName(name);
+		course.setDescription(description);
+		course.setCategories(categories);
+		course.setContent(content);
+
+		when(courseResourceMock.createCourse(course)).thenReturn(new ResponseEntity<>(HttpStatus.CREATED));
+		
+		assertEquals(HttpStatus.CREATED, courseResourceMock.createCourse(course).getStatusCode());
+	}
+	
+	@Test
+	public void insertCourseWhenAddingUnexistingCategories() {
+		String name = "A course title";
+		String description = "A course description";
+		Integer[] categories = {1,2,3};
+		String content = "A course content";
+
+		CourseRequest course = new CourseRequest();
+		course.setName(name);
+		course.setDescription(description);
+		course.setCategories(categories);
+		course.setContent(content);
+		
+		assertEquals(HttpStatus.CONFLICT, courseResource.createCourse(course).getStatusCode());
+	}
+	
+	@Test
+	public void insertCourseWhenAddingARepeatedName() {
+		String name = "A course";
+		String description = "A course description";
+		Integer[] categories = {1,2,3};
+		String content = "A course content";
+
+		CourseRequest course = new CourseRequest();
+		course.setName(name);
+		course.setDescription(description);
+		course.setCategories(categories);
+		course.setContent(content);
+		
+		assertEquals(HttpStatus.CONFLICT, courseResource.createCourse(course).getStatusCode());
+	}
+	
+	@Test
+	public void insertCourseWithEmptyCategories() {

```

Comment:
Nice way of organizing. I prefer using the AAA pattern, separating in arrange, act and assert the test. This way the test is a little bit more structured and easy to understand.